source/troubleshooting/connection-errors.txt

```diff
@@ -0,0 +1,311 @@
+#. Click the :guilabel:`IP Whitelist` menu option.
+
+#. If your IP address is not in the whitelist, click
+   :guilabel:`Add IP Address`.
+
+#. Click :guilabel:`Add Current IP Address`.
+
+#. Click :guilabel:`Confirm`.
+
+|service| recognizes which entries in the whitelist include your
+current IP address, and appends these addresses with
+:guilabel:`(includes your current IP address)`.
+
+.. seealso::
+
+   For more information on configuring whitelist entries, see
+   :atlas:`Configure Whitelist Entries </security-whitelist/>` in the
+   |service| documentation.
+
+Not Master or Secondary is Not Writable
+---------------------------------------
+
+This error message indicates that you have connected to one of the
+secondary members of a replica set without providing the correct
+replica set name *and* requiring a connection to a primary in your
+:guilabel:`Read Preference`.
+
+Solutions
+~~~~~~~~~
+
+Verify Atlas Replica Set Name
+`````````````````````````````
+
+You should ensure that you are providing the correct replica set name
+in the |compass-short| connect dialog.
+
+If your cluster is hosted in |service|, you can obtain the correct
+replica set name using the following procedure:
+
+.. include:: /includes/fact-get-replica-set-name-steps.rst
+
+Allow Compass to Connect to the Secondary
+`````````````````````````````````````````
+Another approach to resolving this error is to allow |compass-short| to
+connect to a secondary cluster member. Change your
+:guilabel:`Read Preference` to either :guilabel:`Primary Preferred` or
+:guilabel:`Secondary Preferred` in your connection settings. This
+should allow you to bypass this error and connect.
+
+.. note::
+
+   When you connect to a secondary cluster member, you cannot
+   create or modify documents, indexes, or databases. Only use this
+   solution when you do not need to perform any write operations.
+
+.. seealso::
+
+   For more information on read preferences, see
+   :manual:`Read Preference </core/read-preference/>` in the MongoDB
+   manual.
+
+Authentication Failed
+---------------------
+
+When you attempt to connect to your cluster using |compass-short|, you
+may encounter an error message stating
+:guilabel:`Authentication Failed`.
+
+Likely Causes
+~~~~~~~~~~~~~
+
+This error message is often the result of one of five conditions:
+
+- Your :guilabel:`Password` is missing after autofilling your
+  connection details from a copied connection string. |compass-short|
+  requires you to manually enter your password in the connection dialog
+  as a security precaution.
+
+- Your :guilabel:`Username` and/or :guilabel:`Password` are incorrect.
+
+- You have not :atlas:`configured any MongoDB users
+  </security-add-mongodb-users/>` for your |service| cluster.
+
+- You did not specify the correct :guilabel:`Authentication Database`.
+  This setting corresponds to the :manual:`authSource
+  </reference/connection-string/#urioption.authSource>` connection
+  string option.
+
+  .. note::
```

Comment:
Not sure a note is needed for this. I think we can do 1 of 2 things:

- Move bullet point 4 (and its accompanying note) to be bullet point 5 (so that we don't chop bullet point 5 away from the list, as it currently looks)

- Remove the Note and keep everything else as-is

Reply:
Yep I'm fine with that. As I look at it again, I think it just doesn't look apparent that the note is associated with bullet point 5 (arguably). I think what makes it potentially confusing is that we use the word "this" in the note, so the context may be lost if someone doesn't know that the note corresponds to bullet point 5 (i.e., "What is 'this'?")

Reply:
Agreed. Will update to keep the context.